compiler/src/dotty/tools/dotc/core/Symbols.scala

```diff
@@ -437,7 +437,7 @@ object Symbols {
 
     /** Does this symbol come from a currently compiled source file? */
     final def isDefinedInCurrentRun(implicit ctx: Context): Boolean = {
-      pos.exists && defRunId == ctx.runId
+      pos.exists && defRunId == ctx.runId && sourceFile == ctx.source.file
```

Comment:
That's not the right condition anymore. See for instance how it is used in `checkNoConflict`. We want to flag top-level definitions in other source files if they clash with a definition in the current source file.

Reply:
We could probably use as third criterion:

     { val file = assocatedFile
       file != null && !file.path.endsWith(".class")
     }

Reply:
Hmm that's tricky because associatedFile forces things, I'll give it a try.